packages/network-enablement-controller/src/NetworkEnablementController.ts

```diff
@@ -0,0 +1,324 @@
+import { BaseController } from '@metamask/base-controller';
+import type {
+  ControllerGetStateAction,
+  ControllerStateChangeEvent,
+  RestrictedMessenger,
+} from '@metamask/base-controller';
+import { BuiltInNetworkName, ChainId, toHex } from '@metamask/controller-utils';
+import type { MultichainNetworkControllerGetStateAction } from '@metamask/multichain-network-controller';
+import type {
+  NetworkControllerGetStateAction,
+  NetworkControllerNetworkAddedEvent,
+  NetworkControllerNetworkRemovedEvent,
+  NetworkControllerStateChangeEvent,
+} from '@metamask/network-controller';
+import type { CaipChainId, CaipNamespace, Hex } from '@metamask/utils';
+import { KnownCaipNamespace, parseCaipChainId } from '@metamask/utils';
+
+import { POPULAR_NETWORKS } from './constants';
+import { SolScope } from './types';
+import { deriveKeys, isOnlyNetworkEnabledInNamespace } from './utils';
+
+const controllerName = 'NetworkEnablementController';
+
+/**
+ * Information about an ordered network.
+ */
+export type NetworksInfo = {
+  /**
+   * The network's chain id
+   */
+  networkId: CaipChainId;
+};
+
+/**
+ * A map of enabled networks by namespace and chain id.
+ */
+type EnabledMap = Record<CaipNamespace, Record<CaipChainId | Hex, boolean>>;
```

Comment:
Since we end up converting chain IDs into CAIP chain IDs whenever we enable and disable networks, can we always store networks under their CAIP chain IDs instead of allowing them to be stored under the hex chain ID? I feel like that would make things easier (and we can always convert back to a hex chain ID if we really need to).

``` suggestion
 * A map of enabled networks by CAIP namespace and CAIP chain ID.
 */
type EnabledMap = Record<CaipNamespace, Record<CaipChainId, boolean>>;